### PR TITLE
Fix resource_id parsing to accept 'resourcegroups'

### PR DESCRIPTION
**Pull request by brendandburns**
Fixes https://github.com/Azure/azure-cli/issues/2248

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/commands/arm.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
 logger = azlogging.get_az_logger(__name__)
 
-regex = re.compile('/subscriptions/(?P<subscription>[^/]*)/resourceGroups/(?P<resource_group>[^/]*)'
+regex = re.compile('/subscriptions/(?P<subscription>[^/]*)/resource[gG]roups/(?P<resource_group>[^/]*)'
```
Comment (vishrutshah) [inline src/azure-cli-core/azure/cli/core/commands/arm.py]: Question: Should we have full case-insensitive regex instead just `resourceGroup` with g / G?

Comment (vishrutshah) [inline src/azure-cli-core/azure/cli/core/commands/arm.py]: CI is failing due to line length more than 100 char.

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_api_check.py [hunk 2]
```diff
@@ -29,6 +29,11 @@ def setUp(self):
     def tearDown(self):
         pass
 
+
+    def test_parse_resource(self):
+        parts = parse_resource_id('/subscriptions/00000/resourcegroups/bocconitestlabrg138089/providers/microsoft.devtestlab/labs/bocconitestlab/virtualmachines/tasktest1')
```
Comment (vishrutshah) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_api_check.py]: Do we also need test with `G` in `resourceGroup` or it's already covered in other test?

File: src/azure-cli-core/azure/cli/core/commands/arm.py (hunk 1 above)
Comment (brendandburns) [inline src/azure-cli-core/azure/cli/core/commands/arm.py]: I'd rather not. Making it case-insensitive here is kind of a hack b/c that's what arm explorer is using. This is a case-by-case workaround, not a case desensitizing in general.

Comment (brendandburns) [inline src/azure-cli-core/azure/cli/core/commands/arm.py]: fixed.

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_api_check.py (hunk 2 above)
Comment (brendandburns) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_api_check.py]: added.

**Comment by brendandburns** [comment]
Comments addressed, please re-check.

Thanks!

**Pull request merged**